src/components/HomePage/URLSearchParamsParser.js

```diff
@@ -0,0 +1,46 @@
+import { useMemo, useEffect } from "react";
+
+import qs from "qs";
+import { useLocation } from "react-router";
+import { useSelector } from "react-redux";
+
+import useOffersSearcher from "./SearchResultsArea/SearchResultsWidget/useOffersSearcher";
+import { SearchResultsConstants } from "./SearchResultsArea/SearchResultsWidget/SearchResultsUtils";
+
+const URLSearchParamsParser = ({ showSearchResults }) => {
+
+    const location = useLocation();
+
+    const offerSearch = useSelector((state) => state.offerSearch);
+    const validQueryParams = Object.keys(offerSearch);
+
+    const queryParams = useMemo(() => qs.parse(location.search, {
+        ignoreQueryPrefix: true,
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Same question as above

Reply:
This as the same thought behind as the code above: we only need this to run once, so we should not have any dependencies.